### ISSUE TITLE
0.22.2 break code generated by swagger

### ISSUE THREAD

**Issue by cedric-appdirect** [go-openapi/validate#160]
When using swagger with JSON object that have a minimum value of `0`, you end up with a structure that contain a pointer to your value (which can be zero and valid). When using 0.22.2, this result in an error as the test case below will reproduce.

```go
func TestValues_RequiredValidZero(t *testing.T) {
	var err error

	m := struct {
		Percent *int64
	}{
		Percent: new(int64),
	}

	err = Required("percent", "body", m.Percent)
	assert.NoError(t, err)
}
```

I am suspecting line 170 in value.go introduced by this recent commit https://github.com/go-openapi/validate/commit/348543c76e9210df64cdda6b5a2a99a6df80b628#diff-a9025362efd5a648c10a6bf6ad93e8a7f729eb991a8fb0dd589928d0018921a2R170 from @jiuker . I don't have enough context to propose a correction, but let me know how I can help.

I would think this is a rather important problem to fix as any swagger generated code that pick this new version will get broken.

**Comment by jiuker** [comment]
https://github.com/go-openapi/validate/pull/159

**Comment by cedric-appdirect** [comment]
Indeed, that's a duplicate. You could include the test case in the repo after merging #159 .

**Comment by fredbi** [comment]
fixed by reverting this commit in v0.22.3
